spec/services/notifications/journal_wp_notification_service_spec.rb

```diff
@@ -221,13 +221,23 @@ def call
         ]
       end
 
-      # Event creation will be prevented by the service
+      it_behaves_like 'creates no notification'
+    end
+
+    context 'assignee has all in app notifications enabled but only involved for mail' do
+      let(:recipient_notification_settings) do
+        [
+          FactoryBot.build(:mail_notification_setting, involved: true),
+          FactoryBot.build(:in_app_notification_setting, involved: false, watched: false, mentioned: false, all: true)
+        ]
+      end
+
       it_behaves_like 'creates notification' do
-        let(:event_reason) { :involved }
+        let(:event_reason) { :subscribed }
```

Comment:
Again wording of `event_reason` and `subscribed`. 
```suggestion
        let(:expected_notification_reason) { :all }
```

Reply:
Renamed in #9430